src/types.jl

```diff
@@ -282,24 +282,10 @@ end
 
 # Equality
 ==(a::ZonedDateTime, b::ZonedDateTime) = a.utc_datetime == b.utc_datetime
+isequal(a::ZonedDateTime, b::ZonedDateTime) = isequal(a.utc_datetime, b.utc_datetime)
+hash(zdt::ZonedDateTime, h::UInt) = hash(zdt.utc_datetime, h)
```

Comment:
Do we want `hash(DateTime(...)) == hash(ZonedDateTime(..., tz"UTC"))`?

Reply:
Hmmm, that's a good point. Maybe we should do `hash(zdt.utc_datetime, hash(:ZonedDateTime, h))`?

Reply:
That'll make our `LaxZonedDateTime` code a little awkward as we'll want a `LaxZonedDateTime` to hash to the same value, but that's probably a reasonable compromise.